RELEASE.md

```diff
@@ -0,0 +1,18 @@
+# Release Process
+
+1. Send announcement that a release is underway to the committer's lounge on
+discourse.numenta.org and ask reviewers not to merge PRs in NuPIC until you're
+done with the release.
+2. Create a PR that includes:
+    - Release notes added to CHANGELOG.md
+    - Change to the VERSION file so it matches the intended release version
+3. Wait for the PR to be approved and merged, and the Bamboo build to complete
+successfully
+4. Create a "release" in Bamboo with a version matching the intended release
```

Comment:
You have to create a "release" before you can deploy. The version you specify there is an arbitrary string (with a simple mechanism that bumps the last number as the default next version) and that version does become available as a deployment variable. So we have to create the Bamboo release and I don't know of a way to enforce that the version matches something from the build so we have to validate that in the deployment.